### DISCUSSION TITLE
POST_FLAT_FILE_FULFILLMENT_DATA in JSON

### DISCUSSION THREAD

**Discussion by Lemodo20** [abuzuhri/Amazon-SP-API-CSharp#794] [General]
So far, I have been using the Fika API to transfer various data such as stock levels, price updates and tracking data to Amazon. These are either CSV data or XML (e.g. for tracking data). Since this is to be switched off in March (I understand that CSV data will no longer be possible), I am now looking for a way to do this via JSON. I have already created a corresponding function for generating JSON. However, I am currently failing with the corresponding transfer calls. I've tried several, but either an XML is expected or Amazon issues an error message. For the tracking data in XML I have so far used POST_FLAT_FILE_FULFILLMENT_DATA. Is there an equivalent for JSon? JSON data does not accept this despite the ContentType.JSON parameter.

**Comment by Lemodo20** [comment]
so nobody have to send json-files to amazon?

**Comment by astanley44** [comment]
See the updated FeedsSample.cs

**Comment by Lemodo20** [comment]
Thank you for the hint. I've tested the example, but get as answer from the API back: "Merchant is needed, but not available".
Don't see this as a parameter till now

also i tried to upload my tracking-Data with this function. I got the message back, that my json-File is not suppored. So how can i upload also my tracking data like this:

    Dim jsonData As String = "
{
  ""header"": {
    ""sellerId"": ""A310xxxCK7xxxx"",
    ""version"": ""2.0""
  },
  ""messages"": [
    {
      ""messageId"": 1,
      ""orderId"": ""xxx-0354503-1742xx"",
      ""trackingNumber"": ""CM487xxxx45DE"",
      ""carrierCode"": ""DHL"",
      ""shipDate"": ""2025-01-31T16:37:26+01:00""
    }
  ]
}"

**Reply by astanley44** [reply]
Use FikaAmazonAPI.AmazonSpApiSDK.Models.Orders;  ConfirmShipmentRequest

**Comment by Lemodo20** [comment]
sorry, but i'm a little bit stupit :-)
can't find a sample for it. so made this:

 amazonConnection = New AmazonConnection(New AmazonCredential() With {
.AccessKey = "AKIAWQQRYCUxxxxx",
.SecretKey = "pi2SgsKRlG2GLS5yxxxxxxxxxxxxx",
.RoleArn = "arn:aws:iam::xxxxx:role/SPRole",
.ClientId = "amzn1.application-oa2-client.xxxxx",
.ClientSecret = secret,
.RefreshToken = "Atzr|IwEBIJQxxxxx",
.MarketPlace = MarketPlace.Germany,
.IsDebugMode = True
})
     Dim confirmShipmentRequest As New ConfirmShipmentRequest()
     Dim packageDetail As New PackageDetail() With {
         .CarrierCode = "12345",
         .CarrierName = "DHL",
         .ShipDate = DateTime.Now,
         .TrackingNumber = "302-3795956-xxxx"
     }

     confirmShipmentRequest.PackageDetail = packageDetail

     Dim orderId As String = "IhreBestellID" ' Setzen Sie hier die tatsächliche Bestell-ID

     Dim result = amazonConnection.Orders.ConfirmShipment(orderId, confirmShipmentRequest)


  | Message | "1 validation error detected: Value '' at 'marketplaceId' failed to satisfy constraint: Member must satisfy regular expression 

don't understand this, because marketplaceID is an value of the FikaAPI.

I can't also find any example for ConfirmShipment.

Maybe you have another hind for me?

**Reply by Lemodo20** [reply]
get ConfirmShipmentRequest now  done, but get this error:

The XML you submitted is ill-formed at the Amazon Envelope XML level at (or near) line 2, column 5.

So it looks like it expect still an XML instead of a JSON, and that is exact, what i want to replace
